src/zcash_script.rs

```diff
@@ -1,16 +1,19 @@
 use std::num::TryFromIntError;
 
-use zcash_primitives::transaction::TxVersion;
-
 use super::interpreter::*;
+use super::script::*;
+use super::script_error::*;
 
 /// This maps to `zcash_script_error_t`, but most of those cases aren’t used any more. This only
 /// replicates the still-used cases, and then an `Unknown` bucket for anything else that might
 /// happen.
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub enum Error {
     /// Any failure that results in the script being invalid.
-    Ok,
+    ///
+    /// __NB__: This is in `Option` because this type is used by both the C++ and Rust
+    ///         implementations, but the C++ impl doesn’t yet expose the original error.
+    Ok(Option<ScriptError>),
```

Comment:
It would be better for now to just use `Ok(ScriptError::UnknownError)` for C++, I think.